### PR TITLE
chore(types): annotate `h3Exports` types

### PR DESCRIPTION
**Pull request by iiio2**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

`resolveModuleExportNames` function returns `Promise<string[]>`. Here we have called it but `h3Exports` is returning `any`. I have explicitly annotated `string[]` for `h3Exports`. Also below we are filtering `h3Exports`. I think `filter` can be used only with arrays. That is why I think we may explicitly annotate `h3Exports` here.  

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/core/config/resolvers/imports.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export async function resolveImportsOptions(options: NitroOptions) {
   options.imports.presets.push(...getNitroImportsPreset());
 
   // Add h3 auto imports preset
-  const h3Exports = await resolveModuleExportNames("h3", {
+  const h3Exports: string[] = await resolveModuleExportNames("h3", {
```
Comment (pi0) [inline src/core/config/resolvers/imports.ts:18]: Type is already inferred: 

image

**Comment by pi0** [comment]
Thanks for PR, but in this case, simply relying on inference should be good enough internally.

File: src/core/config/resolvers/imports.ts (hunk 1 above)
Comment (iiio2) [reply src/core/config/resolvers/imports.ts:18]: oh. i see. thanks.

**Comment by iiio2** [comment]
Ok @pi0 . Thank you so much.

**Pull request closed without merging**